controller/rest/federation.go

```diff
@@ -1599,14 +1643,20 @@ func promoteToMaster(w http.ResponseWriter, acc *access.AccessControl, login *lo
 	if reqData.DeployRepoScanData != nil {
 		cfg.DeployRepoScanData = *reqData.DeployRepoScanData
 	}
-	clusHelper.PutFedSettings(nil, cfg)
+	if err := clusHelper.PutFedSettings(nil, cfg); err != nil {
+		log.WithFields(log.Fields{"error": err}).Error("PutFedSettings")
+	}
 
 	revisions := share.CLUSEmptyFedRulesRevision()
-	clusHelper.PutFedRulesRevision(nil, revisions)
-	clusHelper.PutFedScanRevisions(&share.CLUSFedScanRevisions{ScannedRegRevs: make(map[string]uint64)}, nil)
+	if err := clusHelper.PutFedRulesRevision(nil, revisions); err != nil {
+		log.WithFields(log.Fields{"error": err}).Error("PutImportTask")
+	}
+	if err := clusHelper.PutFedScanRevisions(&share.CLUSFedScanRevisions{ScannedRegRevs: make(map[string]uint64)}, nil); err != nil {
+		log.WithFields(log.Fields{"error": err}).Error("PutImportTask")
+	}
 
 	accFedAdmin := access.NewFedAdminAccessControl()
-	cacheFedEvent(share.CLUSEvFedPromote, msg, login.fullname, login.remote, login.id, login.domainRoles)
+	_ = cacheFedEvent(share.CLUSEvFedPromote, msg, login.fullname, login.remote, login.id, login.domainRoles)
```

Comment:
I can see the error is already logged inside cacheFedEvent().  How about adding some comments, so readers know that it's already handled?

Reply:
added.